GiTinder.Tests/Controllers/SettingsControllerShould.cs

```diff
@@ -0,0 +1,188 @@
+            {
+                HttpContext = httpContext.Object
+            };
+        }
+
+        [Fact]
+        public void ReturnErrorIfTokenInReqHeaderIsEmptyStringInGetSettings()
+        {
+            //Arrange
+            ArrangeForSettingsControllerTests();
+            headerDictionary.Add("X-Gitinder-Token", "");
+
+            //Act
+            var actualResponse = settingsController.GetSettings();
+            var expectedResponseBody = new ErrorResponseBody("Unauthorized request!");
+
+            //Assert
+            Assert.Equal(expectedResponseBody.Status, actualResponse.Status);
+            Assert.Equal(expectedResponseBody.Message, (actualResponse as ErrorResponseBody).Message);
+            httpRequest.VerifyGet(r => r.Headers);
+            httpContext.VerifyGet(a => a.Request);
+            userServices.Verify(s => s.TokenExists(""), Times.Never);
+            httpResponse.VerifySet(r => r.StatusCode = 403);
+        }
+
+        [Fact]
+        public void ReturnErrorIfReqHeaderIsNullInGetSettings()
+        {
+            //Arrange
+            ArrangeForSettingsControllerTests();
+
+            //Act
+            var actualResponse = settingsController.GetSettings();
+            var expectedResponseBody = new ErrorResponseBody("Unauthorized request!");
+
+            //Assert
+            Assert.Equal(expectedResponseBody.Status, actualResponse.Status);
+            Assert.Equal(expectedResponseBody.Message, (actualResponse as ErrorResponseBody).Message);
+            userServices.Verify(s => s.TokenExists(It.IsAny<string>()), Times.Never);
+            httpResponse.VerifySet(r => r.StatusCode = 403);
+        }
+
+        [Fact]
+        public void ReturnErrorIfTokenInReqHeaderNotFoundInGetSettings()
+        {
+            //Arrange
+            ArrangeForSettingsControllerTests();
+            headerDictionary.Add("X-Gitinder-Token", "x");
+            userServices.Setup(s => s.TokenExists(It.IsAny<string>())).Returns(false);
+
+            //Act
+            var actualResponse = settingsController.GetSettings();
+            var expectedResponseBody = new ErrorResponseBody("Unauthorized request!");
+
+            //Assert
+            Assert.Equal(expectedResponseBody.Status, actualResponse.Status);
+            Assert.Equal(expectedResponseBody.Message, (actualResponse as ErrorResponseBody).Message);
+            userServices.Verify(s => s.TokenExists("x"), Times.Once());
+            httpResponse.VerifySet(r => r.StatusCode = 403);
+        }
+               
+        [Fact]
+        public void CallUpdateAndSaveSettingsIfTokenInReqHeaderFoundInPutSettings()
+        {
+            //Arrange
+            ArrangeForSettingsControllerTests();
+            headerDictionary.Add("X-Gitinder-Token", "x");
+            userServices.Setup(s => s.TokenExists(It.IsAny<string>())).Returns(true);
+
+            mockSettings = new Mock<Settings>();
```

Comment:
@FilipTeply instead of creating an empty constructor for the settings you can use the SettingsFactory to create one, and there's no need to mock this object, you can use a real one